pkg/api/v20240812preview/openshiftcluster_validatestatic.go

```diff
@@ -412,3 +420,51 @@ func (sv openShiftClusterStaticValidator) validateDelta(oc, current *OpenShiftCl
 
 	return nil
 }
+
+func (sv openShiftClusterStaticValidator) validatePlatformWorkloadIdentityProfile(path string, pwip *PlatformWorkloadIdentityProfile) error {
+	// PlatformWorkloadIdentityProfile being empty is acceptable
+	if pwip == nil {
+		return nil
+	}
+
+	resourceIDPattern, _ := regexp.Compile("/subscriptions/.*/resourceGroups/.*/providers/.*/.*/.*")
```

Comment:
I think the regex-based validation is fine as it is, but we also maintain a parser that converts a resourceID into a typed struct: https://github.com/Azure/ARO-RP/blob/master/pkg/util/arm/resources.go

Reply:
+1 on using the shared code at https://github.com/Azure/ARO-RP/blob/master/pkg/util/arm/resources.go here.